Try2

ustinovsa opened the pull request:
@konivex 
Переработал итератор и вообще класс Fibonacci
Остальные ошибки тоже исправил(включая возвращаемые значения метода getInstance класса RandomArrayFactroy

ustinovsa committed:
tasks 1, 2, 3, 5 completed

ustinovsa committed:
5 tasks completed

ustinovsa committed:
6 tasks ready

ustinovsa committed:
Метод sum() класса Fibonacci работает только для последнего созданного
экземпляра класса

ustinovsa committed:
Работа выполнена.
Экземпляры класса Fibonacci не мешают друг другу.

ustinovsa committed:
Работа выполнена.
Экземпляры класса Fibonacci не мешают друг другу.

ustinovsa committed:
Работа выполнена.
Экземпляры класса Fibonacci не мешают друг другу.
Добавлены поля для хранения информации в экземпляре класса,
Итератор и экземпляры теперь не хранят коллекции.

konivex commented on src/ru/avalon/java/j20/labs/models/Fibonacci.java:
```diff
@@ -1,57 +1,111 @@
+ * Фибоначчи</a>
  */
 public class Fibonacci implements Iterable<Integer> {
 
```
Проблема с мешающими друг другу экземплярами осталась — если я создам два экземпляра с разным quantity, а потом вызову итераторы, то один из них вернёт мне некорректное количество чисел, т.к. они будут использовать общее значение quantity.

konivex commented on src/ru/avalon/java/j20/labs/tasks/Task3.java:
```diff
@@ -34,5 +38,19 @@ public void run() {
+                + System.lineSeparator()
+                + fib2.getSize() + " = " + fib2.getSum()
+                + System.lineSeparator()
+                + fib3.getSize() + " = " + fib3.getSum());
```
Замените все вызовы fib*.getSum() на fib*.sum() и посмотрите на результат работы программы.

konivex reviewed (changes requested):
Исправьте замечание. Жду до вечера.

P.S.: Я пытаюсь добиться от вас не того, чтобы вы посчитали сумму первых N чисел Фибоначчи, не в этом цель, а цель в том, чтобы понять как работать с итераторами — это основная тема задания.

ustinovsa committed:
Убрал модификатор static, чтобы не было общего использования
 поля quantity;
соответственно убрал модификатор static у FibonacciIterator'а
для использования только при создании экземпляра класса Fibonacci. 
Сейчас вроде тот результат, который должен быть.

ustinovsa commented:
@konivex  Итератор должен создаваться для каждого экземпляра класса, соответственно убрал модификаторы static, чтобы не было общего использования статического поля класса.

konivex reviewed (approved):
Зачтено.
